### ISSUE TITLE
[shelly] support for new device Shell Pro 4PM

### ISSUE THREAD

**Issue by MrMontesa** [openhab/openhab-addons#11105]
Shelly is about to release a new/updated version of their devices. The first device supporting the new environment will be Shelly Pro 4 PM.
https://shop.shelly.cloud/shelly-pro-4pm-wifi-smart-home-automation?utm_source=sendinblue&utm_campaign=Shelly_Pro_4PM_winners&utm_medium=email
They created lots of new documentation: https://shelly-api-docs.shelly.cloud/?fbclid=IwAR3X8Erpruxp5w-bqJ4no78btoC1Ig2BXfhRHP6ozYz_lfMHSM5aorsR0w0
Also this announcement on Facebook:
![image](https://user-images.githubusercontent.com/600125/129388962-52f4b1e6-c8a1-4c87-bf72-8342f971f16b.png)

I was able get an early-bird device which will be delivered tomorrow. I'd like to help enabling device support for the new generation devices.
Let me know how I can help supporting the new device. special debug log, etc.

**Comment by MrMontesa** [comment]
Received the device and started playing with it. Its not auto-detected from a THING scan. So tried adding it manually using `thingTypeUID: shelly:shelly4pro`. ang got the following errors in openhab.log
```
2021-08-15 15:01:19.182 [INFO ] [y.internal.handler.ShellyBaseHandler] - shelly4pro-70596c4eb2: Device is password protected, enter correct credentials in thing configuration.
2021-08-15 15:01:19.182 [INFO ] [y.internal.handler.ShellyBaseHandler] - shelly4pro-70596c4eb2: Thing goes OFFLINE: message.offline.conf-error-access-denied
2021-08-15 15:01:19.183 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.IllegalArgumentException: Invalid device name format: shellydevice
        at org.openhab.binding.shelly.internal.discovery.ShellyThingCreator.getThingType(ShellyThingCreator.java:108) ~[?:?]
        at org.openhab.binding.shelly.internal.discovery.ShellyThingCreator.getThingTypeUID(ShellyThingCreator.java:96) ~[?:?]
        at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.changeThingType(ShellyBaseHandler.java:855) ~[?:?]
        at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.isAuthorizationFailed(ShellyBaseHandler.java:842) ~[?:?]
        at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.lambda$0(ShellyBaseHandler.java:163) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
```

Then removed the authentication on the device and tried adding again, but got the same error. I guess this is due to the fact, that binding itself is already set to authenticate since all my devices are password protected access.

**Comment by markus7017** [comment]
The new Pro line brings a complete new API, which is not backward compatible. I did some initial work on this, but it will take time. I also need to think about unified device support, because the Gen 1 devices will be in the market for years and are still receiving updates.

**Comment by TweetsOfNiklas** [comment]
I also just installed a Pro 4 PM and was pretty sad when I noticed that it currently is not supported in Openhab. Is there an estimate when it will be available? Alternatively there is a option for MQTT integration in the interface.

**Comment by htttito** [comment]
My Shelly is also not working...
`2021-12-25 14:44:28.879 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.NullPointerException: null
	at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.initializeThing(ShellyBaseHandler.java:226) ~[?:?]
	at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.lambda$0(ShellyBaseHandler.java:160) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]`

**Comment by dan-m-joha** [comment]
Same here:
2022-01-05 17:52:00.981 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.IllegalArgumentException: Invalid device name format: shellydevice
        at org.openhab.binding.shelly.internal.discovery.ShellyThingCreator.getThingType(ShellyThingCreator.java:108) ~[?:?]
        at org.openhab.binding.shelly.internal.discovery.ShellyThingCreator.getThingTypeUID(ShellyThingCreator.java:96) ~[?:?]
        at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.changeThingType(ShellyBaseHandler.java:855) ~[?:?]
        at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.isAuthorizationFailed(ShellyBaseHandler.java:842) ~[?:?]
        at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.lambda$0(ShellyBaseHandler.java:163) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]

**Comment by Cmachtel** [comment]
Same for me, Any timeline when or how to resolve it ?

**Comment by tjarb** [comment]
Same here, after installing some dimmer 2 modules I installed an 4PM pro device with the thought it would be more reliable with wired ethernet.
But found out it's not supported by the Shelly binding, quite a disappointed :(
Is there a plan/timeline? Can we participate somehow?

**Comment by tjarb** [comment]
I was thinking, you could make a new "pro" binding for the "pro" series?

**Comment by Cmachtel** [comment]
If you want to control them (Not quite as cool ) but mqqt works.
Here the Channel descriptions: 

`UID: mqtt:topic:9162e145cb:fad70a20ca
label: Shelly4PM
thingTypeUID: mqtt:topic
configuration: {}
bridgeUID: mqtt:broker:9162e145cb
channels:
  - id: Relay2
    channelTypeUID: mqtt:switch
    label: Relay2
    description: null
    configuration:
      commandTopic: shelly4pm/rpc
      formatBeforePublish: '{"src":"MQTT","method":"Switch.Set", "params":{"id":2,"on":%s}}'
      stateTopic: shellies/shellypro4pm/events/rpc
      transformationPattern: JSONPATH:$.params.switch:0.output
      off: "false"
      on: "true"
  - id: Relay2_Volt
    channelTypeUID: mqtt:number
    label: Relay2_Volt
    description: null
    configuration:
      unit: V
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:2
      transformationPattern: JSONPATH:$.voltage
  - id: Relay2_Watt
    channelTypeUID: mqtt:number
    label: Relay2_Watt
    description: null
    configuration:
      unit: W
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:2
      transformationPattern: JSONPATH:$.apower
  - id: Relay2_Ampere
    channelTypeUID: mqtt:number
    label: Relay2_Ampere
    description: null
    configuration:
      unit: A
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:2
      transformationPattern: JSONPATH:$.current
  - id: Relay0
    channelTypeUID: mqtt:switch
    label: Relay0
    description: null
    configuration:
      commandTopic: shelly4pm/rpc
      formatBeforePublish: '{"src":"MQTT","method":"Switch.Set", "params":{"id":0,"on":%s}}'
      stateTopic: shellies/shellypro4pm/events/rpc
      transformationPattern: JSONPATH:$.params.switch:0.output
      off: "false"
      on: "true"
  - id: Relay0_Volt
    channelTypeUID: mqtt:number
    label: Relay0_Volt
    description: null
    configuration:
      unit: V
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:0
      transformationPattern: JSONPATH:$.voltage
  - id: Relay0_Watt
    channelTypeUID: mqtt:number
    label: Relay0_Watt
    description: null
    configuration:
      unit: W
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:0
      transformationPattern: JSONPATH:$.apower
  - id: Relay0_Ampere
    channelTypeUID: mqtt:number
    label: Relay0_Ampere
    description: null
    configuration:
      unit: A
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:0
      transformationPattern: JSONPATH:$.current
  - id: Relay1
    channelTypeUID: mqtt:switch
    label: Relay1
    description: null
    configuration:
      commandTopic: shelly4pm/rpc
      formatBeforePublish: '{"src":"MQTT","method":"Switch.Set", "params":{"id":1,"on":%s}}'
      stateTopic: shellies/shellypro4pm/events/rpc
      transformationPattern: JSONPATH:$.params.switch:0.output
      off: "false"
      on: "true"
  - id: Relay1_Volt
    channelTypeUID: mqtt:number
    label: Relay1_Volt
    description: null
    configuration:
      unit: V
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:1
      transformationPattern: JSONPATH:$.voltage
  - id: Relay1_Watt
    channelTypeUID: mqtt:number
    label: Relay1_Watt
    description: null
    configuration:
      unit: W
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:1
      transformationPattern: JSONPATH:$.apower
  - id: Relay1_Ampere
    channelTypeUID: mqtt:number
    label: Relay1_Ampere
    description: null
    configuration:
      unit: A
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:1
      transformationPattern: JSONPATH:$.current
  - id: Relay3
    channelTypeUID: mqtt:switch
    label: Relay3
    description: null
    configuration:
      commandTopic: shelly4pm/rpc
      formatBeforePublish: '{"src":"MQTT","method":"Switch.Set", "params":{"id":3,"on":%s}}'
      stateTopic: shellies/shellypro4pm/events/rpc
      transformationPattern: JSONPATH:$.params.switch:0.output
      off: "false"
      on: "true"
  - id: Relay3_Volt
    channelTypeUID: mqtt:number
    label: Relay3_Volt
    description: null
    configuration:
      unit: V
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:3
      transformationPattern: JSONPATH:$.voltage
  - id: Relay3_Watt
    channelTypeUID: mqtt:number
    label: Relay3_Watt
    description: null
    configuration:
      unit: W
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:3
      transformationPattern: JSONPATH:$.apower
  - id: Relay3_Ampere
    channelTypeUID: mqtt:number
    label: Relay3_Ampere
    description: null
    configuration:
      unit: A
      formatBeforePublish: "%.4f"
      stateTopic: shelly4pm/status/switch:3
      transformationPattern: JSONPATH:$.current
`

**Comment by tjarb** [comment]
@Cmachtel thank you for your suggestion. 
I was trying to simplify my system by moving from zigbee2mqtt to just the shelly binding for shelly devices (incl pro).
I know need mqtt, sql, openhab and zigbee2mqtt, zwave, and was hoping to loose mqtt, zigbee2mqtt and zwave by using shelly ethernet devices.

**Comment by markus7017** [comment]
Check here: https://community.openhab.org/t/shelly-binding/56862/2464?u=markus7017
merged with PR #12336
